src/operators/rename.ts

```diff
@@ -0,0 +1,60 @@
+import { Operator, OperatorOptions, OperatorValidator } from '../operator';
+
+export interface RenameOperatorOptions extends OperatorOptions {
+  properties: { [key: string]: string };
+}
+
+export class RenameOperator implements Operator {
+  static specification = {
+    index: { required: false, type: ['number'] },
+    properties: { required: true, type: ['object'] },
+  };
+
+  readonly index: number;
+
+  readonly properties: object;
+
+  constructor(public options: RenameOperatorOptions) {
+    const { index = 0, properties = {} } = options;
+    this.index = index;
+    this.properties = properties;
+  }
+
+  handleRename(data: { [key: string]: any }) {
+    const props = Object.getOwnPropertyNames(this.properties);
+    for (let i = 0; i < props.length; i += 1) {
+      const key = props[i];
+      const val = data[key];
+      /* eslint-disable no-param-reassign */
+      delete data[key];
+      // @ts-ignore
```

Comment:
I noticed the ignore, which I assume is because the `object` type isn't indexable.  Since the validator expects a `string` as the renamed property, should we then assume that the existing property is also a string?  (At least for type declaration in the TS file.)  The `object` type would then give way to `{ [key:string]: string }`.  It looks like any numeric property would get coerced to a string at runtime.  Or was there a non-string use case in mind?

Reply:
Ok, I've changed the lint specifiers and fixed the typescript complaints. It's ready for another review.